pages/instructorGradebook/instructorGradebook.ejs

```diff
@@ -2,42 +2,110 @@
+                            var list = $('<ul/>')
+                            otherUsers.forEach(uid => {
+                                list.append('<li><small>' + uid + '</small></li>')
+                            })
+                            warning.append(list).appendTo(form)
+                        }
+                        form.append('<p><small>This change will be overwritten if further questions are answered by the student.</small></p>')
+                        var cancel = $('<button type="button" class="btn btn-secondary mr-2">Cancel</button>')
+                        cancel.on('click', function() {
+                            $(that).popover('hide')
+                        })
+                        form.append(cancel)
+                        form.append('<button type="submit" class="btn btn-primary">Change</button>')
+                        form.submit(function(e) {
+                            e.preventDefault();
+                            $.post($(this).attr('action'), $(this).serialize(), function(data) {
+                                data.forEach(score => {
+                                    $('#gradebook-table').bootstrapTable('updateCellByUniqueId', {
+                                        id: score.user_id,
+                                        field: `score_${score.assessment_id}`,
+                                        value: score.score_perc,
+                                    });
+                                })
+                            }, 'json');
+                            $(that).popover('hide');
+                        })
+                        return form
+                    },
+                    title: 'Change total percentage score',
+                    trigger: 'click',
+                }).on('show.bs.popover', function(e) {
+                    $('button.edit-score').not(this).popover('hide');
+                });
+                $('.spinning-wheel').hide();
+            },
+        });
+    });
+    function gradeFormatter(score, row, index, field) {
+        var assessment_instance_id = row[field + '_ai_id'];
+        var other_users = row[field + '_other'];
+        if (score == null) {
+            return "—";
+        } else {
+            var container = $('<div>');
+            var link = $('<a/>').attr('href', `<%= urlPrefix %>/assessment_instance/${assessment_instance_id}`)
+                .text(Math.floor(score) + '%').appendTo(container);
+            if (<%= authz_data.has_instructor_edit %>) {
+                var button = $('<button/>').attr('type', 'button').addClass('btn btn-xs btn-secondary edit-score')
+                    .attr('tabindex', 0).attr('data-ai-id', assessment_instance_id).attr('data-score', score)
+                    .attr('data-other-users', JSON.stringify(other_users))
+                    .append($('<i class="fa fa-edit" aria-hidden="true"></i>'))
```

Comment:
I wonder whether we should switch to https://icons.getbootstrap.com/icons/pencil-square/ at some point? Font Awesome is becoming increasingly annoying with their commercial push.

Reply:
Done.